### PR TITLE
Payments GatewayTimeout logs and metrics

### PR DESCRIPTION
**Pull request by nieznanysprawiciel**
Fixes in separate PR

### CODE REVIEW

File: core/payment/src/api/debit_notes.rs [hunk 1]
```diff
@@ -193,27 +199,45 @@ async fn send_debit_note(
...
                 response::bad_request(&e)
             }
             Err(e) => response::server_error(&e),
         }
     })
-    .await
+    .await;
+
+    timing!(
+        "payment.debit_notes.provider.sent.time",
```
Comment (tworec) [inline core/payment/src/api/debit_notes.rs:236]: 👍

File: core/payment/src/api/debit_notes.rs [hunk 2]
```diff
@@ -193,27 +199,45 @@ async fn send_debit_note(
...
             dao.mark_received(debit_note_id, node_id).await?;
+
             counter!("payment.debit_notes.provider.sent", 1);
             Ok(())
         }
         .await
         {
-            Ok(_) => response::ok(Null),
+            Ok(_) => {
+                log::info!("DebitNote [{}] sent.", path.debit_note_id);
```
Comment (tworec) [inline core/payment/src/api/debit_notes.rs:224]: 👍

File: core/payment/src/api/debit_notes.rs [hunk 3]
```diff
@@ -331,14 +359,24 @@ async fn accept_debit_note(
         }
         .await
         {
-            Ok(_) => response::ok(Null),
+            Ok(_) => {
+                log::info!("DebitNote [{}] accepted.", path.debit_note_id);
```
Comment (tworec) [inline core/payment/src/api/debit_notes.rs:363]: 👍

File: core/payment/src/api/debit_notes.rs [hunk 4]
```diff
@@ -331,14 +359,24 @@ async fn accept_debit_note(
...
                 return response::bad_request(&e);
             }
             Err(e) => return response::server_error(&e),
         }
     })
-    .await
+    .await;
+
+    timing!(
+        "payment.debit_notes.requestor.accepted.time",
```
Comment (tworec) [inline core/payment/src/api/debit_notes.rs:375]: 👍

File: core/payment/src/api/invoices.rs [hunk 5]
```diff
@@ -204,14 +217,20 @@ async fn send_invoice(
         }
         .await
         {
-            Ok(_) => response::ok(Null),
+            Ok(_) => {
+                log::info!("Invoice [{}] sent.", path.invoice_id);
```
Comment (tworec) [inline core/payment/src/api/invoices.rs:221]: 👍

File: core/payment/src/api/invoices.rs [hunk 6]
```diff
@@ -204,14 +217,20 @@ async fn send_invoice(
...
             Err(Error::Rpc(RpcMessageError::Send(SendError::BadRequest(e)))) => {
                 response::bad_request(&e)
             }
             Err(e) => response::server_error(&e),
         }
     })
-    .await
+    .await;
+
+    timing!("payment.invoices.provider.sent.time", start, Instant::now());
```
Comment (tworec) [inline core/payment/src/api/invoices.rs:232]: 👍

File: core/payment/src/api/invoices.rs [hunk 7]
```diff
@@ -257,14 +287,24 @@ async fn cancel_invoice(
         }
         .await
         {
-            Ok(_) => response::ok(Null),
+            Ok(_) => {
+                log::info!("Invoice [{}] cancelled.", path.invoice_id);
```
Comment (tworec) [inline core/payment/src/api/invoices.rs:291]: 👍

File: core/payment/src/api/invoices.rs [hunk 8]
```diff
@@ -257,14 +287,24 @@ async fn cancel_invoice(
...
                 response::conflict(&"Invoice already accepted by requestor")
             }
             Err(e) => response::server_error(&e),
         }
     })
-    .await
+    .await;
+
+    timing!(
+        "payment.invoices.provider.cancelled.time",
```
Comment (tworec) [inline core/payment/src/api/invoices.rs:303]: 👍

File: core/payment/src/api/invoices.rs [hunk 9]
```diff
@@ -372,14 +416,24 @@ async fn accept_invoice(
         }
         .await
         {
-            Ok(_) => response::ok(Null),
+            Ok(_) => {
+                log::info!("Invoice [{}] accepted.", path.invoice_id);
```
Comment (tworec) [inline core/payment/src/api/invoices.rs:420]: 👍

File: core/payment/src/api/invoices.rs [hunk 10]
```diff
@@ -372,14 +416,24 @@ async fn accept_invoice(
...
                 return response::bad_request(&e)
             }
             Err(e) => return response::server_error(&e),
         }
     })
-    .await
+    .await;
+
+    timing!(
+        "payment.invoices.requestor.accepted.time",
```
Comment (tworec) [inline core/payment/src/api/invoices.rs:432]: 👍

**Review by tworec — APPROVED** [review]
so far, so good

**Review by maaktweluit — APPROVED** [review]
LGTM!

**Comment by nieznanysprawiciel** [comment]
This PR is duplicate of https://github.com/golemfactory/yagna/pull/1290
which targets release branch instead.
Changes will be merged back to master together with release branch

**Pull request merged**